aspnetcore/security/app-secrets.md

```diff
@@ -1,125 +1,250 @@
 ```
 
-# [Visual Studio Code](#tab/visual-studio-code/)
+> [!NOTE]
+> You must be in the same directory as the *.csproj* file to run tools defined in the *.csproj* file's `DotNetCliToolReference` elements.
 
-Add `Microsoft.Extensions.SecretManager.Tools` to the *.csproj* file and run [dotnet restore](/dotnet/core/tools/dotnet-restore). You can use the same steps to install the Secret Manager Tool using for the command line.
+## Set a secret
 
-[!code-xml[](app-secrets/sample/UserSecrets/UserSecrets-before.csproj?highlight=10)]
+The Secret Manager tool operates on project-specific configuration settings stored in your user profile. To use user secrets, define a `UserSecretsId` element within a `PropertyGroup` of the *.csproj* file. The value of `UserSecretsId` is arbitrary, but is unique to the project. Developers typically generate a GUID for the `UserSecretsId`.
 
-Test the Secret Manager tool by running the following command:
+::: moniker range="<= aspnetcore-1.1"
+[!code-xml[](app-secrets/samples/1.1/UserSecrets/UserSecrets.csproj?name=snippet_PropertyGroup&highlight=3)]
+::: moniker-end
+::: moniker range=">= aspnetcore-2.0"
+[!code-xml[](app-secrets/samples/2.1/UserSecrets/UserSecrets.csproj?name=snippet_PropertyGroup&highlight=3)]
+::: moniker-end
+
+> [!TIP]
+> In Visual Studio, right-click the project in Solution Explorer, and select **Manage User Secrets** from the context menu. This gesture adds a `UserSecretsId` element, populated with a GUID, to the *.csproj* file. Visual Studio opens a *secrets.json* file in the text editor. Replace the contents of *secrets.json* with the key-value pairs to be stored. For example:
+> [!INCLUDE[secrets.json file](~/includes/app-secrets/secrets-json-file.md)]
+
+Define an app secret consisting of a key and its value. The secret is associated with the project's `UserSecretsId` value. For example, run the following command from the directory in which the *.csproj* file exists:
 
 ```console
-dotnet user-secrets -h
+dotnet user-secrets set "Movies:ServiceApiKey" "12345"
 ```
 
-The Secret Manager tool will display usage, options and command help.
+In the preceding example, the colon denotes that `Movies` is an object literal with a `ServiceApiKey` property.
 
-> [!NOTE]
-> You must be in the same directory as the *.csproj* file to run tools defined in the *.csproj* file's `DotNetCliToolReference` nodes.
+The Secret Manager tool can be used from other directories too. Use the `--project` option to supply the file system path at which the *.csproj* file exists. For example:
+
+```console
+dotnet user-secrets set "Movies:ServiceApiKey" "12345" --project "C:\apps\WebApp1\src\WebApp1"
+```
+
+## Set multiple secrets
+
+A batch of secrets can be set by piping JSON to the `set` command. In the following example, the *secrets.json* file's contents are piped to the `set` command:
+
+```console
+type .\secrets.json | dotnet user-secrets set
```

Comment:
Maybe using `input.json` would make it clearer that you're not talking about the $HOME/.microsoft/usersecrets/id/secrets.json file.

Also, you might also mention this only works on Windows. Use `cat inputs.json` for macOS/Linux